sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1278,6 +1268,13 @@ def post_port_active_apsel_to_db(self, api, lport, host_lanes_mask):
 
         asic_index = self.port_mapping.get_asic_id_for_logical_port(lport)
         intf_tbl = self.xcvr_table_helper.get_intf_tbl(asic_index)
+        if not intf_tbl:
+            self.log_error("Active ApSel db update: intf_tbl not found for {}".format(lport))
```

Comment:
@mihirpat1 is this not a problem? when does the selected application get written to the DB then?

Reply:
@prgeor
If this happens, it is indeed a problem and should be treated as an error case. However, once the TRANSCEIVER_INFO table is created by the xcvrd main process or SfpStateUpdateTask thread, this function should be executed again since CmisManagerTask has a watch on `TRANSCEIVER_INFO` table and intf_tbl should be present by this time.

Also, this (line 1275 in diff) helps in handling the following case
If xcvrd restarts and the the CMIS SM is first triggered due to CONFIG_DB update, the TRANSCEIVER_INFO table will not have the entry for the corresponding logical port (since the port table has not yet been created by the SfpStateUpdateTask thread or xcvrd main thread with all other fields such as transceiver type, serial number vendor PN etc). At this point, if "show interface transceiver eeprom" CLI is invoked, we see a traceback since the `type` field hasn't been populated.
https://github.com/sonic-net/sonic-utilities/blob/fb2e5cda90ced88249e06b18d8c5717a89ff62b9/scripts/sfpshow#L450
At a later point, once CMIS SM is triggered again due to TRANSCEIVER_INFO table being published to `STATE_DB` by the xcvrd main thread, the selected application data will be published to the already existing TRANSCEIVER_INFO table for the corresponding logical port.

Reply:
@mihirpat1 the get_intf_tbl() is getting a handle to the TRANSCEIVER_INFO table, but the error message is for ApSel ...can you correct the error message? Also should this be just a warning?

Reply:
@prgeor I have changed this to log_warning now.
I haven't modified the error message string since
"Active ApSel db update: " -> refers to the function in which this error is seen (i.e. post_port_active_apsel_to_db). This mainly helps in distinguishing similar errors being used in different places (i.e. intf_tbl can be None in more than 1 place)
"intf_tbl not found for {}" -> Lists the failure details